File: CRT CODING PRACTISE/12 May/prg6.c
//#include <stdio.h>

long long sum(long long n) {
    return n * (n + 1) / 2;
}

int main() {
    long long L, R, K;

    scanf("%lld %lld %lld", &L, &R, &K);

    // Total sum
    long long totalSum = sum(R) - sum(L - 1);

    // Multiples of K
    long long start = (L + K - 1) / K;
    long long end = R / K;

    long long count = end - start + 1;
    long long multipleSum = 0;

    if (count > 0) {
        multipleSum = K * (count * (start + end)) / 2;
    }

    printf("%lld\n", totalSum - multipleSum);

    return 0;
}

File: CRT CODING PRACTISE/12 May/prg6.py
def sum_n(n):
    return n * (n + 1) // 2

L, R, K = map(int, input().split())

# Total sum
total_sum = sum_n(R) - sum_n(L - 1)

# Multiples of K
start = (L + K - 1) // K
end = R // K

count = end - start + 1
multiple_sum = 0

if count > 0:
    multiple_sum = K * (count * (start + end)) // 2

print(total_sum - multiple_sum)